### ISSUE TITLE
feature request: warn if `find -exec`/`-execdir`/`-ok` misses brace placeholder `{}`

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3330]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2067
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
find . -maxdepth 1 -type d -exec echo {}        # SC2067
find . -maxdepth 1 -type d -execdir echo {}     # SC2067

find . -maxdepth 1 -type d -exec echo \;        # allowed but {} seems to be missed
find . -maxdepth 1 -type d -execdir echo \;     # allowed but {} seems to be missed

find . -maxdepth 1 -type d -exec echo +         # find: missing argument to `-exec'
find . -maxdepth 1 -type d -execdir echo +      # find: missing argument to `-execdir'
```

`-maxdepth` is not defined in POSIX but used so that find exits immediately.

#### Here's what shellcheck currently says:

```
myscript:2:34: error: Missing ';' or + terminating -exec. You can't use |/||/&&, and ';' has to be a separate, quoted argument. [SC2067]
myscript:3:37: error: Missing ';' or + terminating -exec. You can't use |/||/&&, and ';' has to be a separate, quoted argument. [SC2067]
```

#### Here's what I wanted or expected to see:

As per [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/find.html) (and as implemented in GNU findutils and BSD):
- `-exec command ;` does not require `{}`, but does a user intentionally omit `{}`?
- `-exec command +` requires `{}`

```diff
 myscript:2:34: error: Missing ';' or + terminating -exec. You can't use |/||/&&, and ';' has to be a separate, quoted argument. [SC2067]
 myscript:3:37: error: Missing ';' or + terminating -exec. You can't use |/||/&&, and ';' has to be a separate, quoted argument. [SC2067]
+myscript:5:34: warning: Missing '{}' for '-exec ... ;'. [SCXXXX]
+myscript:6:37: warning: Missing '{}' for '-execdir ... ;'. [SCXXXX]
+myscript:8:34: error: Missing '{}' for '-exec ... +'. [SCXXXX]
+myscript:9:37: error: Missing '{}' for '-execdir ... +'. [SCXXXX]
```

**Comment by e-kwsm** [comment]
POSIX does not specify `{}` is mandatory for `-ok`, but GNU and BSD require it.

**Comment by e-kwsm** [comment]
``find: missing argument to `-exec'``

https://github.com/koalaman/shellcheck/blob/acc19ea3366a04812df10288707c788ca9e5a382/src/ShellCheck/Analytics.hs#L2339
